### DISCUSSION TITLE
Does not work automatically merge pull requests

### DISCUSSION THREAD

**Discussion by peter-hanzo** [ObsidianPublisher/obsidian-github-publisher#117] [Q&A — Plugin] [answered]
Hi :)
I have two storages obsidian. 
The first uses version plugin 4.11.14
The second uses version plugin 5.3.4
No problems with version 4.11.14, but with version 5.3.4 there is a problems
Does not work automatically merge pull requests and the deletion of files also does not work.
What could be the reason? I use same settings in both storages

**Comment by Mara-Li** [comment]
- Autoclean is disabled for fixed behavior and the settings are check obligatory now. 
- For merge request, do you have a error in console ?

**Reply by peter-hanzo** [reply]
no error in console
Pull-Request is created but no merge

**Reply by Mara-Li** [reply]
Did you enabled auto merging?
No merging without error in console is surprising

**Reply by peter-hanzo** [reply]
I see

plugin:obsidian-mkdocs-publisher:53 HttpError: Validation Failed: {"resource":"PullRequest","code":"custom","message":"A pull request already exists for peter-hanzo:WIKI_HANZO-3-17-2023."}
    at eval (plugin:obsidian-mkdocs-publisher:57:21728)
plugin:obsidian-mkdocs-publisher:53 TypeError: Cannot read properties of undefined (reading 'trim')
    at Nn.eval (plugin:obsidian-mkdocs-publisher:57:12860)
    at Generator.next (<anonymous>)
    at eval (plugin:obsidian-mkdocs-publisher:6:833)
    at new Promise (<anonymous>)
    at w (plugin:obsidian-mkdocs-publisher:6:653)
    at Nn.mergePullRequestOnRepo (plugin:obsidian-mkdocs-publisher:57:12778)
    at Nn.eval (plugin:obsidian-mkdocs-publisher:57:13586)
    at Generator.next (<anonymous>)
    at s (plugin:obsidian-mkdocs-publisher:6:691)

**Reply by peter-hanzo** [reply]
I deleted previous branch.
I see now:

plugin:obsidian-mkdocs-publisher:53 HttpError: Validation Failed: {"resource":"PullRequest","code":"custom","message":"A pull request already exists for peter-hanzo:WIKI_HANZO-3-17-2023."}
at eval (plugin:obsidian-mkdocs-publisher:57:21728)
plugin:obsidian-mkdocs-publisher:53 TypeError: Cannot read properties of undefined (reading 'trim')

**Reply by Mara-Li** [reply]
Yeah, you need to delete the PR ?

**Reply by peter-hanzo** [reply]
I deleted everything. I don't have PR and branch.
but that didn't solve the problem

**Reply by Mara-Li** [reply]
Did you try to reload Obsidian and update the plugin using beta ?

**Reply by peter-hanzo** [reply]
Yes, I try reload Obsidian
I use ver. 1.1.16

**Reply by Mara-Li** [reply]
So, when you do a PR what happens in Github? 
Because it seems to be a git bug

**Reply by peter-hanzo** [reply]
version  4.11.14 adds files to MAIN
version 5.3.4 creates a new branch and adds new files there
I have to do PR manually
Both versions have the same settings.

**Reply by Mara-Li** [reply]
It's literally impossible that the old version add directly to main, as I always used a branch to push.

When the PR can't be merged, it's because the difference between the file are "too much"

Maybe I can add an option to force merging?

I need to check if I can do it...

**Reply by Mara-Li** [reply]
Nop, can't force PR. 
What's your settings? Can I get a link to a PR to check the problem with merging?

**Reply by peter-hanzo** [reply]
Maybe I should use version 4.11.14. It works for me without problems.
I sent you an invite to my repo
https://github.com/peter-hanzo/Digital-Garden/pull/18

**Reply by peter-hanzo** [reply]
**I see, now. the PR is closed for an unknown reason**

This pull request is closed, but the WIKI_HANZO-3-19-2023 branch has unmerged commits.

**Reply by Mara-Li** [reply]
Can I see the settings of your plugin ?

**Reply by Mara-Li** [reply]
And please, don't use an old version...

**Comment by peter-hanzo** [comment, marked as answer]
**I found the reason!**
I didn't fill in this field. I wrote "marge" there and after that everything started working.

![image](https://user-images.githubusercontent.com/12713201/226383510-0820db3d-61ef-4976-8650-43cdeb3a2715.png)

**Reply by Mara-Li** [reply]
Oh, normally, there is a default message. Good to know there is a bug here! Will be fixed in next update, thanks :)
